interview-questions.md

```diff
@@ -1,43 +1,101 @@
 # ASSESSMENT 6: Interview Practice Questions
 
-Answer the following questions.
+## 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
+ 
+Your answer: 
+- `rails g migration`
+- `belongs_to: cohort` will need to be added to the Student model
+- `rails db:migrate`
 
-First, without external resources. Challenge yourself to answer from memory.
-
-Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
+Researched answer:
+- `rails g migration AddCohortToStudents cohort:references`
+- check the generated migration file to make sure it contains the `def change`
+    `add_reference :students, :cohort, foreign_key: true`
+- `rails db:migrate` to apply the foreign key `cohort_id` to represent the association between a cohort and a student
 
-1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
+## 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: 
+- create, update, delete or show?
 
 Researched answer:
+- `GET` requires an `:id` to specify which item you'd like to retrieve
+- `PUT/PATCH` also requires an `:id` to specify which item that is needed to send data with the request to change that particular item
+- `DELETE` needs an `:id` to speficy which item needs to be deleted
```

Comment:
RESTful routes can be confusing for sure. The routes requiring params would be show, edit, update, destroy.